### PR TITLE
Fix/lint warning on button test file

### PR DESCRIPTION
**Pull request by paghar**
- [x] I have followed the [Your First Code Contribution section of the Contributing guide](https://github.com/themesberg/flowbite-react/blob/main/CONTRIBUTING.md#your-first-code-contribution)

Summarize the changes made and the motivation behind them.

Reference related issues using `#` followed by the issue number.

If there are breaking API changes - like adding or removing props, or changing the structure of the theme - describe them, and provide steps to update existing code.

### CODE REVIEW

**Comment by paghar** [comment]
@rluders I have changed the type to fix lint warning.
please check that🙏

**Comment by rluders** [comment]
@nigellima WDYT?

File: src/components/Button/Button.spec.tsx [hunk 1]
```diff
@@ -171,7 +171,7 @@ describe('Components / Button', () => {
       it('should render as button `as={null}`', () => {
         render(
           <ul>
-            <Button as={null as any} label="Something or other" />
+            <Button as={null as null} label="Something or other" />
```
Comment (nigellima) [inline src/components/Button/Button.spec.tsx:174]: @paghar is the type casting here necessary as now you upgraded the `as` prop to `as?: T | null`?

Comment (hamed-dehghan) [reply src/components/Button/Button.spec.tsx:174]: @nigellima Should I remove it from the examples?

Comment (paghar) [reply src/components/Button/Button.spec.tsx:174]: @nigellima 
should I remove example from the documentation?

**Comment by paghar** [comment]
@rluders should I close this PR?

**Comment by rluders** [comment]
> @rluders should I close this PR?

I actually missed this one, didn't have time to test it. Maybe @SutuSebastian could also give some feedback here.

File: src/components/Button/Button.spec.tsx (hunk 1 above)
Comment (SutuSebastian) [inline src/components/Button/Button.spec.tsx:174]: `as={null}` is sufficient here

**Comment by SutuSebastian** [comment]
My only question is:
- is this nullable approach just to be able to test that it should render the default html element (in this case `button`)?

File: src/components/Button/Button.spec.tsx (hunk 1 above)
Comment (paghar) [reply src/components/Button/Button.spec.tsx:174]: You are right

**Comment by paghar** [comment]
> My only question is:
> 
>     * is this nullable approach just to be able to test that it should render the default html element (in this case `button`)?

Yes, it seems to be for this reason.

**Comment by SutuSebastian** [comment]
Then we're all good here

**Comment by paghar** [comment]
@rluders

**Pull request merged**